### PR TITLE
Feature auth endpoints improvement

### PR DESCRIPTION
**Pull request by babygoat**

### CODE REVIEW

**Comment by babygoat** [comment]
@nickhsine

File: controllers/account.go [hunk 1]
```diff
@@ -201,3 +213,275 @@ func (mc *MembershipController) RenewJWT(c *gin.Context) (int, gin.H, error) {
...
+	})
+
+	http.SetCookie(c.Writer, &http.Cookie{
+		Name:     "access_token",
+		Value:    accessToken,
+		Domain:   defaultDomain,
+		MaxAge:   accessTokenMaxAge,
+		Secure:   secure,
+		HttpOnly: true,
+	})
```
Comment (nickhsine) [inline controllers/account.go]: `c.SetCookie` directly?

File: controllers/account.go [hunk 2]
```diff
@@ -201,3 +213,275 @@ func (mc *MembershipController) RenewJWT(c *gin.Context) (int, gin.H, error) {
...
+	}}, nil
+}
+
+// Activate - validate the reporter account
+// if validated, then sign in successfully,
+// otherwise, sign in unsuccessfully.
+func (mc *MembershipController) ActivateV2(c *gin.Context) {
+	const errorWhere = "MembershipController.Activate"
+	const userIDMaxAge = 60 * 60 * 24 * 30
+	const accessTokenMaxAge = 60 * 15
```
Comment (nickhsine) [inline controllers/account.go]: will 15 minutes too short?
I think we can increase to 1 week.

Comment (nickhsine) [reply controllers/account.go]: even 1 month.

File: controllers/oauth.go [hunk 3]
```diff
@@ -303,17 +303,12 @@ func (o *OAuth) Authenticate(c *gin.Context) {
...
-		Email: matchUser.Email.String}
-
 	// hours to seconds
 	maxAge := globals.Conf.App.JwtExpiration
-	authResp, _ := json.Marshal(authJSON)
 
 	// set domain to .twreporter.org
 	// so each hostname of [www|support|tsai-tracker].twreporter.org will be applied
-	c.SetCookie("auth_info", string(authResp), maxAge, "/", "."+globals.Conf.App.Domain, secure, false)
+	c.SetCookie("user_id", fmt.Sprint(matchUser.ID), maxAge, "/", "."+globals.Conf.App.Domain, secure, false)
```
Comment (nickhsine) [inline controllers/oauth.go]: `c.SetCookie("user_email", ...)` here.
Frontend needs the user email.

File: controllers/account.go (hunk 1 above)
Comment (nickhsine) [reply controllers/account.go]: c.SetCookie("user_email", ...) here.
Frontend needs the user email.

**Comment by babygoat** [comment]
@nickhsine

File: middlewares/jwt.go [hunk 4]
```diff
@@ -42,6 +41,23 @@ func ValidateUserID() gin.HandlerFunc {
 	}
 }
 
+func ValidateIDToken() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		user := c.Request.Context().Value("user")
+
+		claims := user.(*jwt.Token).Claims
+
+		if err := claims.Valid(); nil != err {
```
Comment (nickhsine) [inline middlewares/jwt.go]: do we need to verify the user_id in the claims with user_id in the request BODY?

Comment (babygoat) [reply middlewares/jwt.go]: Will address in the later patch

**Pull request merged**